sympy/stats/rv.py

```diff
@@ -1081,52 +1083,80 @@ def sample(expr, condition=None, size=(), library='scipy',
-    >>> samp in N.pspace.domain.set # doctest: +SKIP
+    >>> samp = sample(N)
+    >>> samp in N.pspace.domain.set
     True
-    >>> samp = next(sample(N, N>0)) # doctest: +SKIP
-    >>> samp > 0 # doctest: +SKIP
+    >>> samp = sample(N, N>0)
+    >>> samp > 0
     True
-    >>> samp_list = next(sample(N, size=4)) # doctest: +SKIP
-    >>> [sam in N.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    >>> samp_list = sample(N, size=4)
+    >>> [sam in N.pspace.domain.set for sam in samp_list]
     [True, True, True, True]
+
+    >>> sample(N, numsamples=3) # doctest: +SKIP
+    [6.07745952932677, 8.024735984727329, 3.7581166319235315]
+
+    >>> sample(N, size = (2,3)) # doctest: +SKIP
+    array([[5.42519758, 6.40207856, 4.94991743],
+       [1.85819627, 6.83403519, 1.9412172 ]])
+
+    >>> sample(N, size=(3,4), numsamples=2) # doctest: +SKIP
+    [array([[ 0.07664441, -1.42557566,  3.66110395,  5.37576465],
+       [ 0.4993775 ,  7.80504228, -0.22978144,  4.5228755 ],
+       [ 5.91415701,  6.85960422,  9.52447392,  0.63699082]]),
+    array([[-1.89958054,  7.22563251,  1.74164548,  7.39832687],
+       [ 3.6053125 ,  4.83748511,  4.77375537,  0.35377489],
+       [ 5.31794739,  8.57816162,  2.16933622, -6.13675074]])]
+
     >>> G = Geometric('G', 0.5) # Discrete Random Variable
-    >>> samp_list = next(sample(G, size=3)) # doctest: +SKIP
+    >>> samp_list = sample(G, size=3)
     >>> samp_list # doctest: +SKIP
-    array([10,  4,  1])
-    >>> [sam in G.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    [1, 3, 2]
+    >>> [sam in G.pspace.domain.set for sam in samp_list]
     [True, True, True]
     >>> MN = Normal("MN", [3, 4], [[2, 1], [1, 2]]) # Joint Random Variable
-    >>> samp_list = next(sample(MN, size=4)) # doctest: +SKIP
+    >>> samp_list = sample(MN, size=4)
     >>> samp_list # doctest: +SKIP
-    array([[4.22564264, 3.23364418],
-           [3.41002011, 4.60090908],
-           [3.76151866, 4.77617143],
-           [4.71440865, 2.65714157]])
-    >>> [tuple(sam) in MN.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    [array([2.85768055, 3.38954165]),
+     array([4.11163337, 4.3176591 ]),
+     array([0.79115232, 1.63232916]),
+     array([4.01747268, 3.96716083])]
+    >>> [tuple(sam) in MN.pspace.domain.set for sam in samp_list]
     [True, True, True, True]
 
 
     Returns
     =======
 
-    sample: iterator object
-        iterator object containing the sample/samples of given expr
+    sample: float/list/numpy.ndarray
+        one sample or a collection of samples of the random expression
+
+        sample(X) --> returns float object
+        sample(X, size=int) or sample(X, numsamples=int) --> returns list object
+        sample(X, size=tuple) --> returns numpy.ndarray object
+        sample(X, size=tuple, numsamples=int) --> returns list containing numpy.ndarray objects
+        sample(X, size=int, numsamples=int) --> returns list containing numpy.ndarray objects
```

Comment:
I don't think that we need all of these combinations. I would just go with the first 3 and not have the `numsamples` option. Perhaps the `numsamples` option could be provided but only so that it is compatible with sympy 1.8. The docstring could list it as deprecated.

Also is a numpy array always returned? What happens if pymc3 is used?

Reply:
I am trying to fix pymc3 implementation in sample. It seems like `pymc3.sample` takes the size parameter (which is called draws in pymc3) only in `int`. And the sympy implementation transfer the size and numsamples parameters as a tuple to draws which causes a TypeError. Here I have printed them to confirm.
```python
next(x)
size =  (1, 2)
/home/aman/sympy/sympy/stats/rv.py:1574: FutureWarning: In v4.0, pm.sample will return an `arviz.InferenceData` object instead of a `MultiTrace` by default. You can pass return_inferencedata=True or return_inferencedata=False to be safe and silence this warning.
  samps = pymc3.sample(draws=size, chains=1, progressbar=False, random_seed=seed)[:]['X']
Traceback (most recent call last):
```
My solution would be to set a for loop to set up a output in case of a tuple input to size. Also since numsamples does not exist in pymc3, for loop is need to set that up also.

Reply:
I think that with pymc3 the size parameter should always be passed as an int. Only for numpy/scipy should it be possible to pass a tuple.

Reply:
I thought about that too, but also some users might find the size as tuple functionality useful. Possibly a very small amount of users. For code simplicity, it is best to pass a TypeError in case of a tuple.